src/Cake.Core/IO/FileExtensions.cs

```diff
@@ -116,6 +116,24 @@ public static bool IsClrAssembly(this IFile file)
                 return false;
             }
 
+            if (file.Path != null
+                &&
+                  !(
+                    file.Path.FullPath.EndsWith(".dll") || file.Path.FullPath.EndsWith(".exe")
+                    ||
+                    file.Path.FullPath.EndsWith(".sys") || file.Path.FullPath.EndsWith(".tsp")
+                    ||
+                    file.Path.FullPath.EndsWith(".acm") || file.Path.FullPath.EndsWith(".ax")
+                    ||
+                    file.Path.FullPath.EndsWith(".cpl") || file.Path.FullPath.EndsWith(".drv")
+                    ||
+                    file.Path.FullPath.EndsWith(".efi") || file.Path.FullPath.EndsWith(".mui")
+                    ||
+                    file.Path.FullPath.EndsWith(".ocx") || file.Path.FullPath.EndsWith(".scr")))
+            {
+                return false;
+            }
+
```

Comment:
A switch on extension would be more efficient and readable
```suggestion
            // Is known extension?
            switch (file.Path?.GetExtension().ToLowerInvariant())
            {
                case "dll":
                case "exe":
                case "sys":
                case "tsp":
                case "acm":
                case "ax":
                case "cpl":
                case "drv":
                case "efi":
                case "mui":
                case "ocx":
                case "scr":
                case null:
                    break;

                default:
                    return false;
            }

```

Reply:
Started with `if` only for `dylib`s. While I was waiting for review and periodically checking I decided to add all other extensions.